docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -213,6 +213,48 @@ There is also some helpful information in the section on
 Some of the information in these references is out of date.
 ```
 
+(detect-candidates-for-summarization)=
+## Detect candidates for summarization
+
+In a large codebase it can be hard to find all the functions that may be difficult for the Prover.
+A traditional approach would be to run a simple parametric rule to explore all functions in the 
+relevant contracts, and studying resulting potential timeouts. 
+However such an approach prolongs the feedback loop of working with the Prover.
+
+As an alternative approach, the Prover supports an {ref}`overapproximating <glossary>` _auto-summarization_ mode.
+It is based on the idea that internal view or pure functions (in Solidity) that are analyzed
+and found to be heuristically difficult for the Prover can be automatically summarized as `NONDET`, 
+resulting in two positive outcomes:
+1. The run is faster since complex code is summarized early in the Prover's pipeline
+2. The Prover emits the list of _new_ summaries it auto-generated, so that the user can then adapt the list
```

Comment:
the "new" seems more confusing than helping to me -- if it's important (like you want to emphasize that existing summaries have precedence), just the word is not enough, but I feel like you don't need to mention that here

Reply:
also, 1. lacks a `.` in the end

Reply:
I think it's pretty important distinction that some function that was already summarized will not have its summary overridden

Reply:
right -- I like it better with the extra sentence that specifies it a bit more